rfcs/inline_symbols.md

```diff
@@ -0,0 +1,521 @@
+must skip that number of bytes to move beyond the UTF8 bytes that follow it. The reader will then
+be positioned over the annotated value, which it can read or skip as needed.
+
+##### `0xE2`: Multiple inline symbol annotations
+
+```
+            7       4 3       0
+            +---------+---------+
+Value with  |   14    |    2    |
+multiple    +---------+---------+=======================================+
+annotations :     length [VarUInt]                                      :
+            +-----------------------------------------------------------+
+            :     annotation [VarInt + optional UTF8 representation]    : ...
+            +-----------------------------------------------------------+
+            :     value                                                 :
+            +-----------------------------------------------------------+
+```
+
+Unlike the singleton annotation encoding, the encoding for a value with multiple annotations
+includes a `length` field that indicates the number of bytes that will be used to represent
+the sequence of annotations that follow. It does not include the length of the value itself,
+which provides a length encoding of its own.
+
+Annotations in the sequence do not need to be encoded homogenously; writers can write some
+annotations as symbol IDs and others as inline text. (See [Inline symbol
+structs](#0xf4-inline-symbol-structs) for more detail.)
+
+To skip an annotation sequence, the reader must read the `VarUInt` `length` and skip that number of
+bytes. The reader will then be positioned over the annotated value, which it can read or skip as
+needed.
+
+### Combining inline symbols with Ion templates
+
+When combined with [Ion
+Templates](https://github.com/amzn/ion-docs/blob/ion-templates-rfc/rfcs/ion_templates.md#rfc-ion-templates),
+inline symbols can further reduce the number of symbol IDs that must be added to the symbol table.
+	
+Consider a template representing a class called `com.example.project.Quux`:
+
+```js
+$ion_1_1
+
+// Define a template composed of several symbols:
+
+$ion_symbol_table::{
+  templates: [
+    com.example.project.Quux::{ // Template 1, a struct with 3 blanks
+	  foo: {#0},
+	  bar: {#0},
+	  baz: {#0}
+	}
+  ]
+}
+
+// Invoke template 1 with various different parameters
+
+{#1 1 2 3}
+{#1 4 5 6}
+{#1 7 8 9}
+
+// The above invocations are equivalent to:
+
+com.example.project.Quux::{
+  foo: 1,
+  bar: 2,
+  baz: 3,
+}
+com.example.project.Quux::{
+  foo: 4,
+  bar: 5,
+  baz: 6,
+}
+com.example.project.Quux::{
+  foo: 7,
+  bar: 8,
+  baz: 9,
+}
+```
+
+If we use inline symbol definitions for `foo`, `bar`, `baz`, and `com.example.project.Quux` in our
+template definition, we do not need to add them to the symbol table. By allocating a single template
+ID, we can produce a compact representation of a struct composed of several symbols without growing
+the symbol table at all.
```

Comment:
Is `at all` needed here? Can you leave it as `without growing the symbol table.`?